lib/rules/object-curly-newline.js

```diff
@@ -180,7 +180,7 @@ module.exports = {
                         fix(fixer) {
                             return fixer.removeRange([
                                 openBrace.range[1],
-                                first.range[0]
+                                firstTokenOrComment.range[0]
```

Comment:
I don't think this will work correctly if the comment is on the same line as the opening brace:

```js
var foo = { // comment
};

// gets fixed to

var foo = {//comment
};
```

In other words, a fix is applied, but it's ineffective at correcting the newline issue. In cases like that, it seems like it would be better to avoid performing a fix at all (by returning `null` from the fix function).